### PR TITLE
Use `viewcode` extension instead of `linkcode`

### LINKED ISSUE #3189: Broken [Source] links in docs
## 📚 Documentation

Hello, I was browsing the docs, and out of curiosity, I wanted to check the source for `LightningModule`, and so I clicked on the [Source] link of the `forward` method. The link, however, sends to https://github.com/PyTorchLightning/PyTorch-Lightning/blob/pytorch_lightning/core/lightning/LightningModule.py, which is not right, as the repo name is "pytorch-lightning".

I am not familiar with Sphinx, but I checked the docs and the conf file, and I think that the reason can be found here:

https://github.com/PyTorchLightning/pytorch-lightning/blob/ee4eae884b137054e10f21a44f0d1d427405e310/docs/source/conf.py#L59-L71

Since `github-repo` is set to `project`, which is, indeed, `PyTorch-Lightning`, as in the broken link.

If someone more familiar with Sphinx can confirm this, I'd be glad to submit a PR to fix this.

### PR DESCRIPTION
**Pull request by ydcjeff**
## What does this PR do?

- Fixes #3189 , It creates a `_modules` folder after build, link `[source]` to there
- Use `viewcode` instead of `linkcode`
- It works with `automodule`, `autoclass`, `autofunction`, `automethod`

### PR Preview Build

This `source` link works
https://56692-178626720-gh.circle-artifacts.com/0/html/lightning_module.html#training-step

This `source` link doesn't
https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html#training-step

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: requirements/docs.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ pandoc
 docutils
 sphinxcontrib-fulltoc
 sphinxcontrib-mockautodoc
-https://github.com/PyTorchLightning/lightning_sphinx_theme/archive/master.zip#egg=pt-lightning-sphinx-theme
+https://github.com/ydcjeff/lightning_sphinx_theme/archive/method-css.zip#egg=pt-lightning-sphinx-theme
```
Comment (williamFalcon) [inline requirements/docs.txt:9]: we can't use a theme linked to your account... it has to be under PL.
@Borda

Comment (ydcjeff) [reply requirements/docs.txt:9]: @williamFalcon  Yes, I know.
It's for first time CI preview build.

After https://github.com/PyTorchLightning/lightning_sphinx_theme/pull/12 this is merged, I will change back to PL.

Comment (Borda) [reply requirements/docs.txt:9]: merged 🎉

File: requirements/docs.txt [hunk 2]
```diff
@@ -6,7 +6,7 @@ pandoc
 docutils
 sphinxcontrib-fulltoc
 sphinxcontrib-mockautodoc
-https://github.com/PyTorchLightning/lightning_sphinx_theme/archive/master.zip#egg=pt-lightning-sphinx-theme
+https://github.com/PyTorchLightning/lightning_sphinx_theme/archive/method-css.zip#egg=pt-lightning-sphinx-theme
```
Comment (Borda) [inline requirements/docs.txt:9]:
```suggestion
https://github.com/PyTorchLightning/lightning_sphinx_theme/archive/master.zip#egg=pt-lightning-sphinx-theme
```

**Comment by Borda** [comment]
just testing and sure if it is a better way as before it linked to GitHub source and you simply create a perm link...
cc: @awaelchli

**Review by awaelchli — APPROVED** [review]
I have been browsing around a bit on this build and it works great! I think this is a great improvement, to be able to directly jump to the source without leaving the docs page.

**Comment by ydcjeff** [comment]
Could this be merged just before 1.0? would be suitable for 1.0 release.

**Review by rohitgr7 — APPROVED** [review]
LGTM.

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]
ok, it seems to be more robust... :]

**Review by nateraw — APPROVED** [review]
Nice!

**Pull request merged**